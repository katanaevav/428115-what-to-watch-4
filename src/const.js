export const START_URL = `https://4.react.pages.academy`;

export const NO_FILTER = `All genres`;

export const DELAY_BEFORE_START_PREVIEW = 1000;
export const MAX_RENDERED_MOVIES_AT_TIME = 8;

export const MAX_SIMILAR_MOVIES_COUNT = 4;

export const REVIEWS_COLUMNS_COUNT = 2;

export const MIN_VOLUME = 0.0;
export const NO_FULLSCREEN = false;

export const TABS_NAMES = [`Overview`, `Details`, `Reviews`];

export const Screens = {
  MAIN_SCREEN: 0,
  MOVIE_PAGE_SCREEN: 1,
  CINEMA_SCREEN: 2,
  AUTH_SCREEN: 3,
  ADD_REVIEW_SCREEN: 4,
};

export const SavingStatus = {
  SUCCESS: `SUCCESS`,
  FAIL: `FAIL`,
};

export const Tabs = {
  OVERVIEW_TAB: 0,
  DETAILS_TAB: 1,
  REVIEWS_TAB: 2,
};

export const MONTH_NAMES = [
  `January`,
  `February`,
  `March`,
  `April`,
  `May`,
  `June`,
  `July`,
  `August`,
  `September`,
  `October`,
  `November`,
  `December`,
];

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};
