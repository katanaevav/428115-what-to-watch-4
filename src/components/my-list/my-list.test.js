import React from "react";
import renderer from "react-test-renderer";
import MyList from "./my-list.jsx";
import {AuthorizationStatus} from "../../const.js";
import {Router} from "react-router-dom";
import history from "../../history.js";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import NameSpace from "../../reducer/name-space.js";

const mockStore = configureStore([]);

const Movies = [
  {
    id: 0,
    title: `Bohemian rhapsody`,
    smallPoster: `img/bohemian-rhapsody.jpg`,
    genre: `Drama`,
    year: 2018,
    bigPoster: `img/bohemian-rhapsody.jpg`,
    cover: `img/bohemian-rhapsody.jpg`,
    ratingScore: 7.9,
    runTime: 170,
    ratingCount: 284651,
    isFavorite: true,
    backgroundColor: `#AABBCC`,
    video: ``,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    directors: [`Брайан Сингер`],
    starrings: [`Рами Малек, Люси Бойнтон, Гвилим Ли, Бен Харди, Джозеф Маццелло, Эйдан Гиллен, Аллен Лич, Том Холландер, Майк Майерс, Аарон МакКаскер и другие`],
    descriptions: [`Чествование группы Queen, их музыки и их выдающегося вокалиста Фредди Меркьюри, который бросил вызов стереотипам и победил условности, чтобы стать одним из самых любимых артистов на планете. Фильм прослеживает головокружительный путь группы к успеху благодаря их культовым песням и революционному звуку, практически распад коллектива, поскольку образ жизни Меркьюри выходит из-под контроля, и их триумфальное воссоединение накануне концерта Live Aid, ставшим одним из величайших выступлений в истории рок-музыки.`],
  },
  {
    id: 1,
    title: `Moonrise kingdom`,
    smallPoster: `img/moonrise-kingdom.jpg`,
    genre: `Fantasy`,
    year: 2012,
    bigPoster: `img/moonrise-kingdom.jpg`,
    cover: `img/moonrise-kingdom.jpg`,
    ratingScore: 8.9,
    runTime: 170,
    ratingCount: 240,
    isFavorite: true,
    backgroundColor: `#AABBCC`,
    video: ``,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    directors: [`Уэс Андерсон`],
    starrings: [`Брюс Уиллис, Эдвард Нортон, Билл Мюррей, Фрэнсис МакДорманд, Джаред Гилман, Кара Хэйуорд, Боб Бэлабан, Тильда Суинтон, Джейсон Шварцман, Харви Кейтель и другие`],
    descriptions: [`60-е годы XX века. Пара влюблённых подростков, живущих на острове в Новой Англии, убегает из-под присмотра взрослых. Сэм Шакаски — бойскаут, сирота, от которого отказались приемные родители, из-за своего непростого характера ставший изгоем среди других бойскаутов, и Сьюзи Бишоп — замкнутая двенадцатилетняя неуравновешенная девочка, живущая мечтами о волшебных мирах. После обнаружения пропажи местный шериф начинает расследование, а вожатый лагеря бойскаутов организует поисковый отряд.`],
  },
];

it(`Render My list component`, () => {

  const store = mockStore({
    [NameSpace.STATE]: {
      renderedMoviesCount: 2,
    },
  });

  const tree = renderer.create(
      <Provider store={store}>
        <Router history={history}>
          <MyList
            myMovies = {Movies}
            authorizationStatus = {AuthorizationStatus.AUTH}
            avatarUrl = {``}
          />
        </Router>
      </Provider>, {
        createNodeMock: () => {
          return {};
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
