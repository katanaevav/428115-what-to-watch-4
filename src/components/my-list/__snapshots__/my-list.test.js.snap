// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render My list component 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link "
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <div
        className="user-block__avatar"
      >
        <a
          href="/mylist"
          onClick={[Function]}
        >
          <img
            alt="User avatar"
            height="63"
            src=""
            width="63"
          />
        </a>
      </div>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        data-key={0}
      >
        <div
          className="small-movie-card__image"
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <video
            className="player__video"
            poster="img/bohemian-rhapsody.jpg"
            preload="metadata"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/0"
            onClick={[Function]}
          >
            Bohemian rhapsody
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        data-key={1}
      >
        <div
          className="small-movie-card__image"
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <video
            className="player__video"
            poster="img/moonrise-kingdom.jpg"
            preload="metadata"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/1"
            onClick={[Function]}
          >
            Moonrise kingdom
          </a>
        </h3>
      </article>
    </div>
    
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
